src/client/Microsoft.Identity.Client/ManagedIdentity/ServiceFabricManagedIdentitySource.cs

```diff
@@ -45,8 +45,42 @@ public static AbstractManagedIdentity Create(RequestContext requestContext)
             return new ServiceFabricManagedIdentitySource(requestContext, endpointUri, EnvironmentVariables.IdentityHeader);
         }
 
+        internal override Func<HttpClient> ValidateServerCertificateCallback(RequestContext requestContext)
+        {
+            return () =>
+            {
+                if (s_httpClient == null)
+                {
+                    s_httpClient = new HttpClient(ValidateServerCertificate(requestContext.Logger));
+                }
+                return s_httpClient;
+            };
+        }
+
+        internal HttpClientHandler ValidateServerCertificate(ILoggerAdapter logger)
```

Comment:
This method is named as if it were an action, but it is a creator. Maybe `CreateHandlerWithSslValidation`